### DISCUSSION TITLE
change size of colorbar in viz module

### DISCUSSION THREAD

**Discussion by skoudoro** [fury-gl/fury#7] [General]
_From @samuelstjean on February 15, 2016 11:0_

Is there an option to do that? I looked at the docstring of some actors from this example http://nipy.org/dipy/examples_built/viz_bundles.html#example-viz-bundles and there does not seem to be a way to change the size of the bar/numbers. Looks a bit too big in comparison with the bundle, and fiddling with the window size change the number size but not the bar size itself.

![dipy](https://cloud.githubusercontent.com/assets/3030760/13046866/9afcaf4a-d3db-11e5-84e2-4f8820c844a2.png)

_Copied from original issue: nipy/dipy#923_

**Comment by skoudoro** [comment]
_From @arokem on February 15, 2016 16:9_

This seems like a question better discussed on the mailing list.

**Comment by skoudoro** [comment]
_From @samuelstjean on February 15, 2016 16:12_

This also seems like a missing feature/needs better documentation. Maybe it
is even impossible, could not find out.

2016-02-15 17:09 GMT+01:00 Ariel Rokem notifications@github.com:

**Comment by skoudoro** [comment]
_From @arokem on February 15, 2016 16:14_

The point is that if an answer is provided here, and we close this issue, it will be hard for a casual user to ever find this answer again.

**Comment by skoudoro** [comment]
_From @samuelstjean on February 15, 2016 16:20_

I rather saw this as
1. here is how to do it
   1.a. We should edit the doc example so it's easier to find for random people.

or
1. Oh right, that could be useful to have.

**Comment by skoudoro** [comment]
_From @samuelstjean on February 23, 2016 13:9_

It can be done by accessing directly the vtk properties of the bar actor, albeit not super user friendly.

Example

``` python
bar = actor.scalar_bar(lut_cmap)
# bar.SetHeight(0.5)
# bar.SetWidth(0.15)
```

Some position properties also can do that according to the vtk doc.

![dipy](https://cloud.githubusercontent.com/assets/3030760/13252446/fecb342c-da36-11e5-9106-9260c09f13f1.png)

**Comment by dmreagan** [comment]
`actor.scalar_bar` could use a PR to add more customization parameters. We'll want to be careful not to complicate it too much, but we can add a few options like:

- title
- horizontal or vertical
- font size
- bar size

[This blog post](https://blog.kitware.com/color-legend-improvements-coming-to-paraview-5-4/) describes some recent improvements to the ParaView color legends. I hope some of those are in VTK so that we can take advantage of them.

**Comment by Nibba2018** [comment]
Should I add a PR for `horizontal/vertical`, `font-size` and `bar-size` for `actor.scalar_bar`?

**Comment by skoudoro** [comment]
Hi @Nibba2018,

Yes, you can go ahead, just be careful on the design.

**Comment by hnumair** [comment]
Are you still working on this @Nibba2018 ?

**Comment by skoudoro** [comment]
I think you can go ahead @hnumair and fix this issue

@Nibba2018 , If you want to work on something else, you can update RangeSlider like you did for DoubleLineSlider

**Comment by Nibba2018** [comment]
@hnumair No you can go ahead with it.

@skoudoro Ok :+1:
